src/main.rs

```diff
@@ -151,6 +151,29 @@ fn get_input_files(files: Vec<String>) -> anyhow::Result<Vec<String>> {
     return Ok(input.lines().map(|f| f.to_string()).collect());
 }
 
+fn get_input_files(files: Vec<String>) -> anyhow::Result<Vec<String>> {
+    let mut input_files = get_input(files)?;
+    // This is a special case where we want to expand `.` and `..`.
+    let dots = vec![".", ".."];
+    if input_files.len() == 1 && dots.contains(&input_files[0].as_str()) {
+        input_files = expand_dir(&input_files[0])?;
+    }
+    if input_files.is_empty() {
+        return Err(anyhow!("No input files on stdin or as args. Aborting."));
+    }
+
+    Ok(input_files)
+}
+
+fn expand_dir(path: &str) -> anyhow::Result<Vec<String>, io::Error> {
+    Ok(fs::read_dir(path)?
+        .flatten()
```

Comment:
what is this flatten call for?

Reply:
Oh ok, weird indeed, could you maybe use the filter_map instead for reading clarity (in #29  ?)

_I'll try to understand why that happen_

Reply:
Ok I got it, it is because `flatten` calls `iter` on each element if it exists, and [Result has an `iter`](https://doc.rust-lang.org/core/result/enum.Result.html#method.iter)

I do think it's not intuitive in this case, but do as want